core/backend/src/IModelHost.ts

```diff
@@ -412,15 +412,13 @@ export class IModelHost {
       });
 
       if (options.crashReportingConfig.enableNodeReport) {
-        try {
-          // node-report reports on V8 fatal errors and unhandled exceptions/Promise rejections.
-          const nodereport = require("node-report/api"); // eslint-disable-line @typescript-eslint/no-var-requires
-          nodereport.setEvents("exception+fatalerror+apicall");
-          nodereport.setDirectory(options.crashReportingConfig.crashDir);
-          nodereport.setVerbose("yes");
-          Logger.logTrace(loggerCategory, "Configured native crash reporting (node-report)");
-        } catch (err) {
-          Logger.logWarning(loggerCategory, "node-report is not installed.");
+        if (process.report !== undefined) {
+          process.report.reportOnFatalError = true;
+          process.report.reportOnUncaughtException = true;
```

Comment:
should we also set process.signal? is `SIGUSR2` sufficient? 
<img width="515" alt="image" src="https://user-images.githubusercontent.com/11051042/191366903-306ecacd-92e5-41f1-ab3c-5c21c8958ede.png">
<img width="516" alt="image" src="https://user-images.githubusercontent.com/11051042/191366943-8d6163ab-c7be-46f6-8d06-bfecc2feee7d.png">
<img width="175" alt="image" src="https://user-images.githubusercontent.com/11051042/191366980-29289d38-9cac-4193-b0e4-a0cdf21a3040.png">

Reply:
Not sure that it will be used? But I don't see the harm in this, so I decided to just add it.

Reply:
sorry i was hoping @wgoehrig would see and comment. @nick4598 @paulius-valiunas any thoughts here? would any of these be valuable for us?

Reply:
I'm assuming that reportOnFatalError set to true would encompass a lot of the signals we care about / have seen in sentry, but I don't know for sure. I.e. a SIGSEGV in node addon would surface itself anyways since its a fatal error. @mattbjordan can you confirm that using the NativeDevTools? We may have already done this in our call last week, but I can't remember.

Reply:
@nick4598 I can't tell if that is the case. We had trouble getting importing/initializing IModelHost in the node test file we made, so couldn't trigger NativeDevTools. The only significant detail I figured out after messing around with this some more is that killing the node program from the command line with a SIGSEGV signal did not cause a report to be generated (but that makes sense since reportOnSignal was not set).

Also, when running a test where I set up the crashConfig, SIGUSR2 is for some reason detected as invalid: `TypeError: Unknown signal: SIGUSR2`, so I reverted the change for now.

Reply:
Its probably useful to be able to generate a node report on demand through the use of reportOnSignal (something we might as well have in my opinion), but I still don't know what this was used for in the past... @calebmshafer would you know?

I'm sort of surprised that a SIGSEGV isn't considered a fatal error and something that causes node to generate a report, but I guess it doesn't matter since sentry would pick that up anyways. I'm assuming here, that a sentry report is  far more useful for diagnosing a SIGSEGV over a node report, but unsure of the contents of node report.

Reply:
Node.js Diagnostic Report example on exception: https://nodejs.org/api/report.html

Reply:
How about we leave `reportOnSignal` as false for now so that we can get this merged? Then we can test in a dev environment to see when/how the diagnostic reports are triggered and consider if we want to add in a signal trigger.